kuksa-sdk/src/main/kotlin/org/eclipse/kuksa/extension/vss/VssLeafBooleanExtension.kt

```diff
@@ -16,17 +16,16 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-package org.eclipse.kuksa.extension.vssProperty
+package org.eclipse.kuksa.extension.vss
 
-import org.eclipse.kuksa.extension.copy
-import org.eclipse.kuksa.vsscore.model.VssProperty
+import org.eclipse.kuksa.vsscore.model.VssLeaf
 
 /**
  * Convenience operator for [copy] with a [Boolean] value which will be inverted.
  *
  * @throws [IllegalArgumentException] if the copied types do not match.
  * @throws [NoSuchElementException] if no copy method was found for the class.
  */
-operator fun VssProperty<Boolean>.not(): VssProperty<Boolean> {
+operator fun VssLeaf<Boolean>.not(): VssLeaf<Boolean> {
```

Comment:
I think we initially planned to call VssLeaf something like VssSignal? 
I guess you used VssLeaf to stay consistent with VssNode from graph theory terminology? 

Do you think it might make sense to introduce a VssBranch and VssSignal (instead of VssLeaf) which both inherit from VssNode, so we get closer again to the VSS Terminology. 

However in our public APIs we would still have to use VssNode since we don't have any branch specific methods

Reply:
Introduced:
```
VssSignal : VssNode
VssBranch : VssNode
```
Also see #99 for further improvements. #99 should not introduce breaking changes as this ticket does because it just extends the definition with interfaces.